tools/ci-ts/test-helpers/common.test.ts

```diff
@@ -0,0 +1,103 @@
+import { ethers } from 'ethers'
+import { assert } from 'chai'
+import { logEvents } from './common'
+
+jest.spyOn(global.console, 'log')
+
+const mockContractName = 'TEST_CONTRACT'
+
+const mockEvents = [1, 2, 3].map(n => ({
+  topic: `topic${n}`,
+  name: `event${n}`,
+  decode: jest.fn().mockReturnValue({ [`input${n}`]: `value${n}` }),
+  inputs: [{ name: `input${n}` }],
+}))
+
+const mockEventNames = mockEvents.map(e => e.name)
+
+const mockContract = ({
+  interface: {
+    events: {
+      event1: mockEvents[0],
+      event2: mockEvents[1],
+      event3: mockEvents[2],
+    },
+  },
+  on: jest.fn(),
+} as unknown) as ethers.Contract
+
+const mockEventEmission1 = {
+  topics: ['topic1'],
+  data: 'eventEmissionData',
+  blockNumber: 1,
+}
+
+const mockEventEmission2 = {
+  topics: ['topic2'],
+  data: 'eventEmissionData',
+  blockNumber: 1,
+}
+
+afterEach(() => {
+  ;(console.log as jest.Mock).mockReset()
```

Comment:
I think we can avoid casting here by assigning the spy return value to our own variable. `const mockLog = jest.spyOn(global.console, 'log')`